### PR TITLE
[WIP] enh: turn interface into callable

### PR DESCRIPTION
**Pull request by satra**
turning interfaces into a callable to support dask based operations.

### CODE REVIEW

**Comment by oesteban** [comment]
I think this should work in nipype 1.0 without much trouble. Therefore, I see under 1.0 anything that makes the jump between 1.0 and 2.0 easier but does not mean to actually implement 2.0.

**Comment by effigies** [comment]
Anything left to this one?

File: nipype/interfaces/base/core.py [hunk 1]
```diff
@@ -587,6 +587,11 @@ def run(self, cwd=None, ignore_exception=None, **inputs):
 
         return results
 
+    def __call__(self, **kwargs):
+        """Make interface into a callable function"""
+        results = self.run(**kwargs)
+        return results.outputs
```
Comment (satra) [inline nipype/interfaces/base/core.py]: what about this to match with the call to `run`?

```suggestion
        return results
```

Comment (effigies) [reply nipype/interfaces/base/core.py]: I have zero expectations of behavior here, so sure?

Comment (oesteban) [reply nipype/interfaces/base/core.py]: Dask might expect something like `results.outputs`.

**Comment by effigies** [comment]
Hi, just a reminder that the 1.1.8 release is targeted for January 28. Please let us know if you'd like to try to finish this up for that release.

**Comment by effigies** [comment]
This pull request is "orphaned," which means it has been deemed to be abandoned by its original author. Orphaned pull requests have not been rejected, and we hope that if a user sees one that will meet their needs with a little work, that they will fork it and open a new pull request (or, in the case of the original author, reopen the original PR).

We ask that all adopted PRs be updated to merge or rebase the current master. If you would like to adopt a PR and need help getting started, any of a number of contributors will be happy to help.

**Pull request closed without merging**